source/controllers/users.ts

```diff
@@ -1,85 +1,61 @@
-import { Request, Response, NextFunction } from 'express';
-import Logging from '../configs/logging';
-import Bcryptjs, { hash } from 'bcryptjs';
-import Mongoose from 'mongoose';
-import Users from '../models/users';
-import SignJwt from '../functions/sign-JWT'
-
-
-
 
-const NAMESPACE = 'User';
-const validateToken = (req: Request, res: Response, next: NextFunction) => {
-    Logging.info(NAMESPACE, 'Token validated');
 
-    return res.status(200).json({
-        message: 'Authorized'
-    });
-};
+import Bcryptjs from 'bcryptjs';
+import Mongoose from 'mongoose';
+import Users from '../models/users';
+import SignJwt from '../functions/sign-jwt';
 
-const addUser = (req: Request, res: Response, next: NextFunction) => {
-    let { username, password } = req.body;
 
-    Bcryptjs.hash(password, 10, (hashError, hash) => {
-        if (hashError) {
-            return res.status(500).json({
-                message: hashError.message,
-                error: hashError
-            });
-        }
+const addUser = async(ctx ,next) => {
+    let { username, password } = ctx.request.body;
 
-        //insert user to DB
-        const user = new Users({
-            _id: new Mongoose.Types.ObjectId(),
-            username,
-            password:hash
-        })
+    const user = new Users({
+        _id: new Mongoose.Types.ObjectId(),
+        username,
+        password: await Bcryptjs.hash(password, 10)
+    })
 
-        return user.save()
-        .then(user => {
-            return res.status(201).json({
-                user
-            })
-        })
-        .catch(error => {
-            return res.status(500).json({
-                message:error.message,
-                error
-            }) 
-        })
-    });
+    const find = await Users.find({username})  
```

Comment:
use `UserModel.findOne({ username })` if you want to find only one document and use the data. If your concern is only about the existence then use `UserModel.exists({ username })`.